adafruit_bme680.py

```diff
@@ -403,6 +482,180 @@ def _read(self, register: int, length: int) -> bytearray:
     def _write(self, register: int, values: bytearray) -> None:
         raise NotImplementedError()
 
+    # garberw added begin ===========================
+    def set_gas_heater(self, heater_temp: UINT16, heater_time: UINT16) -> bool:
+        """
+        *  @brief  Enable and configure gas reading + heater
+        *  @param  heater_temp
+        *          Desired temperature in degrees Centigrade
+        *  @param  heater_time
+        *          Time to keep heater on in milliseconds
+        *  @return True on success, False on failure
+        """
+        if (heater_temp == 0) or (heater_time == 0):
+            return False
+        # enable = BME68X_ENABLE
+        try:
+            self._set_heatr_conf(heater_temp, heater_time)
+        except GasHeaterException:
+            return False
+        return True
+
+    def _set_heatr_conf(self, heater_temp: UINT16, heater_time: UINT16) -> None:
+        # restrict to BME68X_FORCED_MODE
+        op_mode: UINT8 = _BME68X_FORCED_MODE
+        # restrict to enable = True
+        enable: bool = True
+        nb_conv: UINT8 = 0
+        hctrl: UINT8 = _BME68X_ENABLE_HEATER
+        run_gas: UINT8 = 0
+        ctrl_gas_data_0: UINT8 = 0
+        ctrl_gas_data_1: UINT8 = 0
+        ctrl_gas_addr_0: UINT8 = _BME68X_REG_CTRL_GAS_0
+        ctrl_gas_addr_1: UINT8 = _BME68X_REG_CTRL_GAS_1
```

Comment:
I think this is using a lot of local variables where using what they equate to might be more efficient.  This comment holds for the other methods as well.